### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by jonbgallant**
Restructured the readme to put the links to sdks, libraries and samples at the top and made a bunch of edits to make it easier to digest.

 

# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Comment by BryanLa** [comment]
Hi @jonbgallant - @jspaith asked me to have a look at this. Looks good, only one small suggestion to change `https://docs.microsoft.com/en-us/azure/iot-dps/` to `https://docs.microsoft.com/azure/iot-dps/` Thanks..

**Comment by jonbgallant** [comment]
@BryanLa Thanks for the review.  I updated the URL. Cheers, Jon

File: readme.md [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
-* **Microsoft Azure IoT Hub service SDK for C** to interface with an Azure IoT Hub service instance from a back-end C application
-* **Serializer library for C** to help serialize and deserialize data on your device.
+* **Azure IoT Hub Device C SDK** to connect devices running C code to Azure IoT Hub.
+* **Azure IoT Hub Device Provisioning Service Client SDK** for enrolling devices with [Azure IoT Device Provisioning Services](https://docs.microsoft.com/azure/iot-dps/) and managing enrollments lists.
+* **Azure IoT Hub Service C SDK** to interface with an Azure IoT Hub service instance from a server-side C application.
+* **Serializer Library for C** to help serialize and deserialize data on your device.
+
+## Packages and Libraries
+  The simplest way to use the Azure IoT SDKs is to use the following packages and libraries:
+  * Linux (Ubuntu 14.04, 15.04, 15.10, 16.04): [Device SDK on apt-get](./iothub_client/readme.md#aptgetpackage)
```
Comment (jspaith) [inline readme.md]: We currently only build for 14.04 and 16.04.  How many / far back to support is an ongoing discussion but for now please remove 15.04 and 15.10.

**Comment by jonbgallant** [comment]
@jspaith  - I removed 15.04 and 15.10

File: readme.md [hunk 2]
```diff
@@ -1,42 +1,45 @@
...
-* **Microsoft Azure Iot Hub Device Provisioning Service client SDK** for enrolling devices with Device Provisioning Services and managing enrollments lists.
-* **Microsoft Azure IoT Hub service SDK for C** to interface with an Azure IoT Hub service instance from a back-end C application
-* **Serializer library for C** to help serialize and deserialize data on your device.
+* **Azure IoT Hub Device C SDK** to connect devices running C code to Azure IoT Hub.
+* **Azure IoT Hub Device Provisioning Service Client SDK** for enrolling devices with [Azure IoT Device Provisioning Services](https://docs.microsoft.com/azure/iot-dps/) and managing enrollments lists.
+* **Azure IoT Hub Service C SDK** to interface with an Azure IoT Hub service instance from a server-side C application.
+* **Serializer Library for C** to help serialize and deserialize data on your device.
+
+## Packages and Libraries
+  The simplest way to use the Azure IoT SDKs is to use the following packages and libraries:
```
Comment (jspaith) [inline readme.md]: >use [](start = 22, length = 3)

Please replace with "get started with".  Subtle, but a lot of folks ultimately want their fork for productization for total control or debugging.

**Comment by jonbgallant** [comment]
@jspaith - Should be good now.

**Pull request merged**